bundles/org.dataflowanalysis.analysis/src/org/dataflowanalysis/analysis/dsl/constraint/DSLDestinationSelector.java

```diff
@@ -0,0 +1,26 @@
+package org.dataflowanalysis.analysis.dsl.constraint;
+
+import org.dataflowanalysis.analysis.dsl.AnalysisConstraint;
+
+/**
+ * Represents the top level DSL object to constrain destination vertices of a constraint
+ */
+public class DSLDestinationSelector {
+    private final AnalysisConstraint analysisConstraint;
+
+    /**
+     * Creates a new DSL destination selector with the given analysis constraint
+     * @param analysisConstraint Given analysis constraint
+     */
+    public DSLDestinationSelector(AnalysisConstraint analysisConstraint) {
+        this.analysisConstraint = analysisConstraint;
+    }
+
+    /**
+     * Constrains attributes of the destination vertex of the constraint
+     * @return Returns destination node selector DSL object
+     */
+    public DSLNodeDestinationSelector toVertex() {
```

Comment:
Currently, `toVertex` is the only possbile destination. I guess this is what you meant with the missing possibility to evaluate joined streams? So that `toData` could be added later? If this is correct, it would be fine for me to add this later (but we should document is somewhere) in order to speed up the merge of this giant PR!

Reply:
Your observation is correct, that `toVertex` is the only destination right now.
However, the writing flow of the DSL in my opinion should be thus:
`<....>.neverFlows().toVertex()`.
This class enforces this syntax.
Furthermore, as you've pointed out this would allow us to add further types, like `toData`, in the future

Reply:
I completely agree, then let's keep it as it is and add `toData` to future improvements issue